carnival/carnival.ts

```diff
@@ -0,0 +1,1017 @@
+
+namespace SpriteKind {
+    //% isKind
+    export const Ball = SpriteKind.create()
+    //% isKind
+    export const Booth = SpriteKind.create()
+    //% isKind
+    export const Mouse = SpriteKind.create()
+    //% isKind
+    export const Crosshair = SpriteKind.create()
+    //% isKind
+    export const Moon = SpriteKind.create()
+}
+
+enum winTypes {
+    //% block="win game"
+    Win,
+    //% block="lose game"
+    Lose,
+    //% block="best score"
+    Score,
+    //% block="best time"
+    Timed,
+    //% block="multiplayer"
+    Multi,
+    //% block="custom" blockHidden
+    Custom
+}
+
+enum scoreTypes {
+    //% block="high score"
+    HScore,
+    //% block="low score"
+    LScore,
+    //% block="high time"
+    HTime,
+    //% block="low time"
+    LTime,
+    //% block="none"
+    None
+}
+
+enum speeds {
+    //% block="fast"
+    Fast,
+    //% block="medium"
+    Med,
+    //% block="slow"
+    Slow
+}
+
+enum areas {
+    //% block="top"
+    Top,
+    //% block="middle"
+    Mid,
+    //% block="bottom"
+    Bottom
+}
+
+enum tracers {
+    //% block="full"
+    Full,
+    //% block="partial"
+    Part,
+    //% block="pointer"
+    Pointer,
+    //% block="crosshair"
+    Cross,
+    //% block="off"
+    Off
+}
+
+let textSprite: TextSprite = null
+//let fanfare: effects.BackgroundEffect = undefined;
+//let winStyle="winTypes.Score"
+
+// Get array of player info
+let players: info.PlayerInfo[];
+
+
+
+/**
+* An extension full of carnival goodness
+*/
+//% weight=100 color=#b70082 icon="\uf54e"
+//% groups='["Ball", "Timer", "Countdown", "Game", "Scene"]'
+namespace carnival {
+
+    /**
+    * Adds text to the top, middle, or bottom
+    * of screen as defined by games
+    */
+    //% group="Scene"
+    //% blockId=add_label_to
+    //% block="add label $myLabel to $myPosition of window || $myColor"
+    //% myLabel.defl="Whack-the-Mole"
+    //% myColor.shadow="colorindexpicker"
+    //% myColor.defl=4
+    //% myPosition.defl=areas.Bottom
+    //% inlineInputMode=inline
+    //% help=github:carnival/docs/add_label_to
+    export function add_label_to(myLabel: string, myPosition: areas, myColor?: number) {
```

Comment:
javascript/typescript convention is to use camel case, not snake case. Rename to `addLabelTo`

Reply:
Fixed